sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -136,10 +137,7 @@ private[sql] object OrcFilters {
     case FloatType | DoubleType =>
       value.asInstanceOf[Number].doubleValue()
     case _: DecimalType =>
-      val decimal = value.asInstanceOf[java.math.BigDecimal]
-      val decimalWritable = new HiveDecimalWritable(decimal.longValue)
-      decimalWritable.mutateEnforcePrecisionScale(decimal.precision, decimal.scale)
```

Comment:
why do we remove this line?

Reply:
This will potentially result in a Exception in the hashCode method of HiveDecimalWritabble.

Reply:
Can you provide more details? I'm not convinced that we can skip it.

Reply:
https://github.com/apache/orc/blob/d1d2a37842087501933d594a71cd1f5f5c4a5a9e/java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java#L784 

There are similar conversions in the ORC project.

@dongjoon-hyun Why do we need mutateEnforcePrecisionScale ?

Reply:
@cloud-fan I have an ORC file to reproduce the RuntimeException with mutateEnforcePrecisionScale. But have not come up with a nice unit test yet.

Reply:
@sadhen and @cloud-fan .
Yes, Line 140 was the bug and `mutateEnforcePrecisionScale` just amended the scale and precision.